cpp/src/gandiva/cache.h

```diff
@@ -25,6 +25,14 @@
 
 namespace gandiva {
 
+namespace internal {
+// Only called once by GetCapacity().
+// Do the actual work of getting the capacity from env var.
+// Also makes the testing easier.
+GANDIVA_EXPORT
+int GetCapacityFromEnvVar();
+}  // namespace internal
+
 GANDIVA_EXPORT
 int GetCapacity();
```

Comment:
Can this be called `GetCacheCapacity`? The current name is too imprecise.

Reply:
Yes, it can. But is this public API?

Reply:
I have no idea :-) You could deprecate the old API if we want to ensure a smoother migration.

Reply:
Addressed with deprecating old API and adding renamed one.